File: src/world.cpp
#include "math.h"
#include "src.h"

// Appending new objects to the world
void World::add_body(Body &body) { this->bodies.push_back(body); }

// Generating friction
void World::generate_friction() {
  for (int i = 0; i < this->bodies.size(); i++) {
    this->bodies[i].apply_force(this->bodies[i].vel.get().multi(-1).multi(0.5));
    if (this->bodies[i].vel.x < 0.3 && this->bodies[i].vel.x > -0.3) {
      this->bodies[i].vel.x = 0;
    }
    if (this->bodies[i].vel.y < 0.3 && this->bodies[i].vel.y > -0.3) {
      this->bodies[i].vel.y = 0;
    }
  }
}

File: go_src/world.go
package src

// Physical World Class

type World struct {
	Bodies []Body
}

func NewWorld() *World {
	return &World{}
}

// Appending new objects to the world
func (world *World) AddBody(body *Body) {
	world.Bodies = append(world.Bodies, *body)
}

// Generating Friction
func (world *World) GenerateFirction() {
	for i := range world.Bodies {
		world.Bodies[i].ApplyForce(*world.Bodies[i].Vel.Get().Multi(-1).Multi(0.5))
		if world.Bodies[i].Vel.X < 0.3 && world.Bodies[i].Vel.X > -0.3 {
			world.Bodies[i].Vel.X = 0
		}
		if world.Bodies[i].Vel.Y < 0.3 && world.Bodies[i].Vel.Y > -0.3 {
			world.Bodies[i].Vel.Y = 0
		}
	}
}
